### PR TITLE
update 3d viewer, and json-dataset 

### PR DESCRIPTION
**Pull request by toloudis**
Problem
=======
Internal scientists are again requesting to use onprem datasets.  So we need to make sure the json-dataset code is working.  Also discovered that the current 3d viewer was out of sync and was failing to load volume data.

Solution
========
Update the onprem datastore.
"http://dev-aics-dtp-001.corp.alleninstitute.org/cfedata/cell-feature-data/datasets.json" ( true location is //allen/aics/animated-cell/Allen-Cell-Explorer/cell-feature-data/datasets.json ) now must hold a list of subdirectories which have datasets in the format prescribed in allen-cell-animated/cell-feature-data repo.

Currently CFE is updated to only support the "single dataset" version and megasets are not yet being handled.   Most of the interesting changes are in `json-dataset/index.ts`.
Also issued new 3d viewer releases and updated 3d viewer version to confirm that data still loads.

This is phase 1.  Phase 2 would be to add support for megasets, but it is likely that for internal use the presentation of megasets is not yet important.

Steps to Verify:
----------------
http://dev-aics-dtp-001.corp.alleninstitute.org/cell-feature-explorer/dist/index.html
now points to this branch, which loads raw datasets from the above mentioned onprem directory (the 3d image data is still in cloud)

### CODE REVIEW

File: src/state/image-dataset/json-dataset/index.ts [hunk 1]
```diff
@@ -92,31 +89,111 @@ class JsonRequest implements ImageDataset {
...
+                            megasetInfo.publications = megaset.publications || [];
+                            megasetInfo.extra = megaset.extra || "";
+                            const id = `${topLevelJson.name}_v${topLevelJson.version}`;
+                            megasetInfo.name = id;
+                            // for single dataset sets we want to store the document with the whole id, to avoid
+                            // grouping versions of the same name together as if they're megasets
+                            megasetInfo.datasets = {
+                                [id]: {
+                                    ...topLevelJson,
+                                    image: `${datasetdirpath}/${topLevelJson.image}`,
```
Comment (toloudis) [inline src/state/image-dataset/json-dataset/index.ts:131]: built-in assumption that the image file is local to the current dataset.json file

File: src/state/image-dataset/json-dataset/index.ts [hunk 2]
```diff
@@ -92,31 +89,111 @@ class JsonRequest implements ImageDataset {
...
-            .then((metadata: AxiosResponse) => metadata.data);
+        //const megasets: Megaset[] = [];
+        return axios.get(`${this.listOfDatasetsDoc}`).then((metadata: AxiosResponse) => {
+            const datadir = this.listOfDatasetsDoc.substring(
+                0,
+                this.listOfDatasetsDoc.lastIndexOf("/")
+            );
+
+            return Promise.all(
+                (metadata.data as string[]).map((datasetdir) => {
```
Comment (frasercl) [inline src/state/image-dataset/json-dataset/index.ts:107]: Both the top-level function and the argument to `map` could become `async`. By no means necessary, but would save a couple indents.

File: src/state/image-dataset/json-dataset/index.ts [hunk 3]
```diff
@@ -92,31 +89,111 @@ class JsonRequest implements ImageDataset {
...
+                            console.log("has datasets; unexpected; dropping");
+                            return {} as Megaset;
+                        } else {
+                            const topLevelJson = { ...megaset } as unknown as DatasetMetaData;
+                            const megasetInfo = { ...megaset };
+                            megasetInfo.title = topLevelJson.title;
+                            megasetInfo.publications = megaset.publications || [];
+                            megasetInfo.extra = megaset.extra || "";
+                            const id = `${topLevelJson.name}_v${topLevelJson.version}`;
+                            megasetInfo.name = id;
```
Comment (frasercl) [inline src/state/image-dataset/json-dataset/index.ts:125]: This block starts with three copies of the same object (`megaset`, `megasetInfo`, `topLevelJson`) and copies properties between them.

1. I see `megasetInfo` is eventually returned from here, and it looks like `topLevelJson` is there to keep an unmodified copy around to copy into `megasetInfo.datasets` below. Could `megaset` be copied in directly?
2. Looks like the title assignment on 121 may be redundant.
3. Suggestion: some of the remaining property assignments which just apply defaults could happen at assignment of `megasetInfo`, e.g.
```javascript
const megasetInfo = {
    publications: megaset.publications || [],
    extra: megasetInfo.extra || "",
    // ...
    ...megaset
};
```

Comment (toloudis) [reply src/state/image-dataset/json-dataset/index.ts]: I believe this is now addressed. It is much simpler now.

File: src/state/image-dataset/firebase/index.ts [hunk 4]
```diff
@@ -90,7 +90,7 @@ class FirebaseRequest implements ImageDataset {
             });
     };
 
-    public setCollectionRef = (id: string) => {
+    private setCollectionRef = (id: string) => {
```
Comment (toloudis) [inline src/state/image-dataset/firebase/index.ts:93]: the only substantive change in this file is to change this from public to private.  The rest of the file just got auto-formatted.

File: src/state/image-dataset/json-dataset/index.ts (hunk 2 above)
Comment (toloudis) [reply src/state/image-dataset/json-dataset/index.ts:109]: I'm going to hold off on this change as the diff is already a bit much.

Comment (toloudis) [reply src/state/image-dataset/json-dataset/index.ts]: agreed, I went ahead and fixed that

**Review by ShrimpCryptid — APPROVED** [review]
Thank you for your code comments!

**Review by frasercl — APPROVED** [review]

**Pull request merged**